DQMServices/FwkIO/plugins/DQMRootSource.cc

```diff
@@ -533,7 +555,17 @@ std::shared_ptr<edm::FileBlock> DQMRootSource::readFile_() {
     if (!isGoodFile && m_skipBadFiles)
       continue;
 
-    m_openFiles.insert(m_openFiles.begin(), file);
+    std::unique_ptr<std::string> guid{file->Get<std::string>(kCmsGuid)};
+
+    edm::JobReport::Token jrToken;
+    bool tokenValid{guid};
+
+    if (tokenValid) {
+      edm::Service<edm::JobReport> jr;
+      jrToken = jr->inputFileOpened(
+          pfn, lfn, std::string(), std::string(), "DQMRootSource", "source", *guid, std::vector<std::string>());
```

Comment:
I believe in principle we could report also the open/closes of the "old files" (that have UUID only via TFile), but I also suppose it wouldn't really matter in practice (assuming merging/harvesting old DQMIO files would be rare activity).

Reply:
It also simplifies the code, cleaning up some parts that were a bit ugly, so I've implemented this and tested that it does the right thing with an input missing the CMS GUID